akka-docs/src/main/paradox/additional/rolling-updates.md

```diff
@@ -39,10 +39,33 @@ Additionally you can find advice on @ref:[Persistence - Schema Evolution](../per
 
 ## Cluster Sharding
 
-During a rolling update, sharded entities receiving traffic may be moved during @ref:[shard rebalancing](../typed/cluster-sharding-concepts.md#shard-rebalancing), 
-to an old or new node in the cluster, based on the pluggable allocation strategy and settings.
-When an old node is stopped the shards that were running on it are moved to one of the
-other old nodes remaining in the cluster. The `ShardCoordinator` is itself a cluster singleton. 
+During a rolling update, sharded entities receiving traffic may be moved, based on the pluggable allocation
+strategy and settings. When an old node is stopped the shards that were running on it are moved to one of the
+other remaining nodes in the cluster when messages are sent to those shards.
+
+To make rolling updates as smooth as possible there is a configuration property that defines the version of the
+application. This is used by rolling update features to distinguish between old and new nodes. For example,
+the default `LeasShardAllocationStrategy` avoids allocating shards to old nodes during a rolling update.
+The `LeasShardAllocationStrategy` sees that there is rolling update in progress when there are members with
+different configured `app-version`.
+
+To make use of this feature you need to define the `app-version` and increase it for each rolling update.
+
+```
+akka.cluster.app-version = 1.2.3
+```
+
+To understand which is old and new it compares the version numbers using normal conventions,
+see @apidoc[akka.util.Version] for more details.
+
+Rebalance is also disabled during rolling updates, since shards from stopped nodes are anyway supposed to be
+started on new nodes.
```

Comment:
Perhaps note that messages to a shard that was on the old node and was stopped is allocated as needed rather than waiting until rebalances start again?

Reply:
added one more sentence